docs: add SSR caveat to Ionic Framework docs (#1066)
analogjs/analog: apps/docs-app/docs/integrations/ionic/index.md

  selector: 'demo-root',
  standalone: true,
  imports: [IonApp, IonRouterOutlet],

>   template: `<ion-app><ion-router-outlet></ion-router-outlet></ion-app>`,

Reviewer: @eduardoRoth is this all you did to get routing to work?

I tried replicating your instructions with the `mobile-client` app in this repo https://github.com/RyanClementsHax/nx-storybook-repro/tree/ionic-analog, but get the following error

```
main.ts?t=1715210251271:5 ERROR Error: NG04014: Invalid configuration of route ''. One of the following must be provided: component, loadComponent, redirectTo, children or loadChildren
    at validateNode (chunk-O34KBEB7.js?v=ac462775:2099:13)
    at validateConfig (chunk-O34KBEB7.js?v=ac462775:2046:5)
    at chunk-O34KBEB7.js?v=ac462775:3366:56
    at chunk-RF5WUHVG.js?v=ac462775:1816:31
    at OperatorSubscriber2._this._next (chunk-RF5WUHVG.js?v=ac462775:597:9)
    at Subscriber2.next (chunk-RF5WUHVG.js?v=ac462775:300:12)
    at chunk-RF5WUHVG.js?v=ac462775:1950:20
    at OperatorSubscriber2._this._next (chunk-RF5WUHVG.js?v=ac462775:597:9)
    at Subscriber2.next (chunk-RF5WUHVG.js?v=ac462775:300:12)
    at Observable2._subscribe (chunk-RF5WUHVG.js?v=ac462775:1400:18)
```

I'm trying to solve it on my own, but wanted to see if you encountered this.

Author: Hi @RyanClementsHax , that's everything I did. I'll take a look at your brach later tonight, or maybe tomorrow morning to see if there's something different.
You can create a new repo with my template and also check what's different there: https://github.com/eduardoRoth/analog-ionic-capacitor-template

Reviewer: Thanks a bunch. I'll take a look!

Reviewer: I found the problem. I exported the page component as `export class HomeComponent` instead of `export default class HomeComponent`

Author: Awesome! Those pesky minor details 😂
Glad that you found it fast

@import '@ionic/angular/css/palettes/dark.system.css';
```

### Server Side Rendering Caveat

> Ionic Framework [doesn't support the Angular's new Client Hydration](https://github.com/ionic-team/ionic-framework/issues/28625#issuecomment-1843919548), as `@angular/ssr` [doesn't support web components](https://github.com/angular/angular/issues/52275), and when they are supported, work has to be done on the Stencil components to enable it. So right now there are three ways to handle this:

Reviewer 2: We don't use the Angular SSR package, so this could be a little more clear as an Angular limitation.

```suggestion
Ionic Framework [doesn't support the Angular's new Client Hydration](https://github.com/ionic-team/ionic-framework/issues/28625#issuecomment-1843919548), as `Angular` [doesn't support web SSR with components](https://github.com/angular/angular/issues/52275), and when they are supported, work has to be done on the Stencil components to enable it. So right now there are three options to handle this:
```

Author: Updated with your comments @brandonroberts

> Ionic Framework [doesn't support the Angular's new Client Hydration](https://github.com/ionic-team/ionic-framework/issues/28625#issuecomment-1843919548), as Angular [doesn't support SSR with web components](https://github.com/angular/angular/issues/52275), and when they are supported, work has to be done on the Stencil components to enable it. So right now there are three options to handle this:

Reviewer 3: ```suggestion
Ionic Framework [doesn't support Angular's new Client Hydration](https://github.com/ionic-team/ionic-framework/issues/28625#issuecomment-1843919548), as Angular [doesn't support SSR with web components](https://github.com/angular/angular/issues/52275), and when they are supported, work has to be done on the Stencil components to enable it. So right now, there are three options to handle this:
```

Author: @ajitzero updated.

   export const appConfig: ApplicationConfig = {
     providers: [
       provideFileRouter(),
       //provideClientHydration(), // remove this.
       provideHttpClient(withFetch()),
       { provide: RouteReuseStrategy, useClass: IonicRouteStrategy },
       provideIonicAngular(),
     ],
   };
   ```

2. Add `ngSkipHydration` attribute to the `ion-app` tag.

   - This will disable the client hydration mechanism for the `ion-app` element and children, but will continue to use client hydration on other elements. This will also cause a flicker in the page for the Ionic components. This is not that helpful for other elements/components as, with Ionic apps, all your Ionic components exist inside the `ion-app` tag.

     ```ts
     import { Component } from '@angular/core';
     import { IonApp, IonRouterOutlet } from '@ionic/angular/standalone';

     @Component({
       selector: 'demo-root',
       standalone: true,
       imports: [IonApp, IonRouterOutlet],
       template: `
         <ion-app ngSkipHydration>
           <ion-router-outlet></ion-router-outlet>
         </ion-app>
       `,
     })
     export class AppComponent {}
     ```

3. Disable SSR completely

>    - Disable SSR in the `vite.config.ts` file. This will disable SSR in your whole app but **will eliminate the flickering**.

Reviewer 3: ```suggestion
   - Disable SSR in the `vite.config.ts` file. This will disable SSR in your whole app and **will eliminate the flickering**.
```
minor: "but" makes it sound as if eliminating the flicking is bad.

Author: @ajitzero Rephrased it to give emphasis on what is the disadvantages of disabling it.

     ```ts
     plugins: [
       analog({
         ssr: false,
       }),
     ],
     ```

You **must** pick one of the previous options, as not configuring this will make your app throw errors on runtime, like the following:

```js
ERROR Error: NG0500: During hydration Angular expected <ion-toolbar> but found a comment node.

Angular expected this DOM:

  <ion-toolbar color="secondary">…</ion-toolbar>  <-- AT THIS LOCATION
  …

Actual DOM is:

<ion-header _ngcontent-ng-c1775393043="">
  <!--  -->  <-- AT THIS LOCATION
  …
</ion-header>

> Note: attributes are only displayed to better represent the DOM but have no effect on hydration mismatches.

Reviewer 3: ```suggestion
:::note

Attributes are only displayed to represent the DOM better and do not affect hydration mismatches.

:::
```

This syntax adds a note block as per Docusaurus. Also made minor capitalization/grammar change, "but"=>"and"

Author: @ajitzero 
This error is taken exactly from the DevTools console, as Angular generates it.
